wandb/util.py

```diff
@@ -1554,9 +1554,13 @@ def _is_databricks() -> bool:
     return False
 
 
-def _is_py_or_dockerfile(path: str) -> bool:
+def _is_py_requirements_or_dockerfile(path: str) -> bool:
     file = os.path.basename(path)
-    return file.endswith(".py") or file.startswith("Dockerfile")
+    return (
+        file.endswith(".py")
+        or file.startswith("Dockerfile")
+        or file == "requirements.txt"
```

Comment:
why do you need it? we always send `requirements.txt` see here: https://github.com/wandb/wandb/blob/main/wandb/sdk/wandb_run.py#L2434

Reply:
Thats a fair point, but given that the requirements.txt are saved to runs and this would instead capture the requirements.txt in the code artifact I think this is a better solution. For 2 reasons:
First, right now, in the Launch SDK we never have to query for runs associated with artifacts. So going from code artifact -> associated runs -> a file that exists on the run is a lot less efficient.

Second, if a version of an important package changes between 2 runs, this is not currently captured by our code artifacts. So I'd have one code artifact representing to explicitly defined environments, which one should we use when building the environment? Having this explicitly captured in the code artifact makes it easier.

An alternative, would be to have the job capture any requirements.txt that exists in the code. I'm fine with that as a solution. But I think it's worth thinking about: does the `requirements.txt` capture important information for users when thinking about their code (as I think one would be while looking at a code artifact). Nonetheless, if you prefer I think I can go the route of capturing this in the job artifact as opposed to the code artfiact.

Reply:
No strong feelings here. I was just trying to understand the change. Thanks for the detailed explanation. 

The only thing I don’t like is that we will have 3 places where we have the requirement file. 
- just on the run 
- In the job artifact 
- And now also on the code artifact 


Also I would think a more useful or additional info is to have actually the loaded packages as these are the one needed for the execution of the code. Obviously strictly a subset of the requirements content.